### PR TITLE
Upgrade sentry sdk

### PR DESCRIPTION
**Pull request by victoriachan**
Ticket: https://projects.torchbox.com/projects/support-team/tickets/218

Upgrade deprecated raven sdk to use newer sentry-sdk.

### CODE REVIEW

File: tbx/settings/base.py [hunk 1]
```diff
@@ -322,6 +317,48 @@
...
+    if "SENTRY_ENVIRONMENT" in env:
+        sentry_kwargs.update({"environment": env["SENTRY_ENVIRONMENT"]})
+
+    release = get_default_release()
+    if release is None:
+        try:
+            # But if it's not, we assume that the commit hash is available in
+            # the GIT_REV environment variable. It's a default environment
+            # variable used on Dokku:
+            # http://dokku.viewdocs.io/dokku/deployment/methods/git/#configuring-the-git_rev-environment-variable
```
[the remark this answers is not in the document]
Comment (victoriachan) [reply tbx/settings/base.py:348]: Thanks for reviewing this, Dan. This is a good point. I thought heroku might still use `env["GIT_REV"]` but when I checked some newer projects, they are falling back on `env["HEROKU_RELEASE_VERSION"]` instead. 

I copied this from wagtail kit. Shall I remove these lines, including `SENTRY_RELEASE = env["GIT_REV"]`, from wagtail kit as well then?

[the remark this answers is not in the document]
Comment (zerolab) [reply tbx/settings/base.py:348]: I can't recall the last time I saw `GIT_REV` in Heroku.
We use the dyno metadata https://devcenter.heroku.com/articles/dyno-metadata#dyno-metadata which uses `HEROKU_RELEASE_VERSION`

So yes to cleaning it all down to 
```
            try:
                # Assume this is a Heroku-hosted app with the "runtime-dyno-metadata" lab enabled
                release = env["HEROKU_RELEASE_VERSION"]
            except KeyError:
                # If there's no commit hash, we do not set a specific release.
                release = None
```

[the remark this answers is not in the document]
Comment (victoriachan) [reply tbx/settings/base.py]: Updated this now, after updating the same on wagatial kit.

File: poetry.lock [hunk 2]
```diff
@@ -289,7 +289,7 @@ python-versions = ">=3.4"
 Django = ">=1.8,<4"
 
 [package.extras]
-lint = ["isort (==4.3.21)", "flake8 (==3.8.3)", "black (==19.10b0)"]
+lint = ["black (==19.10b0)", "flake8 (==3.8.3)", "isort (==4.3.21)"]
```
Comment (zerolab) [inline poetry.lock:292]: note: these type of changes come from mismatch between poetry versions. Should not be an issue though

**Review by zerolab — APPROVED** [review]
LGTM, thank you @victoriachan 

Let's go to staging for a quick test, then live

File: poetry.lock (hunk 2 above)
Comment (victoriachan) [reply poetry.lock:292]: Ah. I always use the poetry within the local docker build, so the version should be consistent. Perhaps it was using a different poetry previously.

**Comment by victoriachan** [comment]
This is on [staging](https://torchbox.staging.torchbox.com/) now. And after running `./manage.py sentry_test`, I got these on Sentry: https://sentry.io/organizations/torchbox/issues/?project=1378654 

There is no `release` on the sentry log. BTW the [get_default_release()](https://github.com/getsentry/sentry-python/blob/b36d84a76bd6f8344c9b0a9694591939296e9c06/sentry_sdk/utils.py#L58) method does look in `HEROKU_SLUG_COMMIT`. But this isn't set on torchbox_staging too. Looks like it didn't have `runtime-dyno-metadata` enabled. 

UPDATE: I have enabled it now. And it is working for the latest test errors.

**Comment by zerolab** [comment]
Would be great to bring this up-to-date and deployed

**Pull request merged**